server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java

```diff
@@ -234,8 +235,10 @@ private static ServerHotspot adaptHotspot(Entity storedHotspot) {
     var startLineOffset = (Integer) storedHotspot.getProperty(START_LINE_OFFSET_PROPERTY_NAME);
     var endLine = (Integer) storedHotspot.getProperty(END_LINE_PROPERTY_NAME);
     var endLineOffset = (Integer) storedHotspot.getProperty(END_LINE_OFFSET_PROPERTY_NAME);
-    var textRange = new org.sonarsource.sonarlint.core.commons.TextRange(startLine, startLineOffset, endLine, endLineOffset);
+    var hash = (String) storedHotspot.getProperty(RANGE_HASH_PROPERTY_NAME);
+    var textRangeWithHash = new TextRangeWithHash(startLine, startLineOffset, endLine, endLineOffset, hash);
```

Comment:
The hash might be null for hotspots that were already in the store, could this cause a problem later?

Reply:
Today the hash won't be used for tracking, as in `ServerHotspotTrackable` we return `null` from `getTextRange()`. As not all hotspots will have a hash (see previous comment) you might need to return something from `ServerHotspotTrackable.getTextRange()` only if a hash is present